File: app/src/main/java/org/amoustakos/boilerplate/util/ValidationUtil.java
package org.amoustakos.boilerplate.util;

import android.text.TextUtils;

import java.net.MalformedURLException;
import java.net.URL;
import java.util.regex.Pattern;

import timber.log.Timber;


public final class ValidationUtil {



    public static boolean isValidEmail(CharSequence target) {
        try {
            return !TextUtils.isEmpty(target) && android.util.Patterns.EMAIL_ADDRESS.matcher(target).matches();
        }catch(Exception e){
            Timber.d(e);
            return false;
        }
    }


    public static boolean isValidURL(String url){
        try {
            new URL(url);
            return true;
        } catch (MalformedURLException e) {
            Timber.d(e);
            return false;
        }
    }


    public static boolean isValidIPv4(String ip){
        try{
            Pattern ipPattern = Pattern.compile(
                    "^(([01]?\\d\\d?|2[0-4]\\d|25[0-5])\\.){3}([01]?\\d\\d?|2[0-4]\\d|25[0-5])$");
            return ipPattern.matcher(ip).matches();
        } catch (Exception e){
            Timber.d(e);
            return false;
        }
    }


    public static boolean isValidPort(String port){
        try{
            int fPort = Integer.parseInt(port);
            return fPort>=1 && fPort<=65535;
        }catch(Exception e){
            Timber.d(e);
            return false;
        }
    }

}

File: app/src/main/java/org/amoustakos/boilerplate/util/ValidationUtil.kt
package org.amoustakos.boilerplate.util

import android.text.TextUtils
import timber.log.Timber
import java.net.MalformedURLException
import java.net.URL
import java.util.regex.Pattern


object ValidationUtil {


    fun isValidEmail(target: CharSequence): Boolean {
        try {
            return !TextUtils.isEmpty(target) && android.util.Patterns.EMAIL_ADDRESS.matcher(target).matches()
        } catch (e: Exception) {
            Timber.d(e)
            return false
        }

    }


    fun isValidURL(url: String): Boolean {
        try {
            URL(url)
            return true
        } catch (e: MalformedURLException) {
            Timber.d(e)
            return false
        }

    }


    fun isValidIPv4(ip: String): Boolean {
        try {
            val ipPattern = Pattern.compile(
                    "^(([01]?\\d\\d?|2[0-4]\\d|25[0-5])\\.){3}([01]?\\d\\d?|2[0-4]\\d|25[0-5])$")
            return ipPattern.matcher(ip).matches()
        } catch (e: Exception) {
            Timber.d(e)
            return false
        }

    }


    fun isValidPort(port: String): Boolean {
        return try {
            val fPort = Integer.parseInt(port)
            fPort in 1..65535
        } catch (e: Exception) {
            Timber.d(e)
            false
        }

    }

}
